testsuite/cluster-test/src/health/mod.rs

```diff
@@ -93,12 +96,24 @@ impl HealthCheckRunner {
         )
     }
 
-    /// Returns list of failed validators
-    pub fn run(&mut self, events: &[ValidatorEvent]) -> Vec<String> {
+    /// Takes a list of affected_validators. If there are validators which failed
+    /// which were not part of the experiment, then it returns an Err with a string
+    /// of all the unexpected failures.
+    /// Otherwise, it returns a list of ALL the failed validators
+    /// It also takes a bool parameter: only_print_on_failure. If this is set
+    /// to true, messages are printed only when there are failures.
+    /// If this is set to false, messages are always printed
+    pub fn run(
+        &mut self,
+        events: &[ValidatorEvent],
+        affected_validators_set: &HashSet<String>,
+        only_print_on_failure: bool,
```

Comment:
@andll : I have set this to true at all call sites. I am not sure at which call sites this should be set to true and where it should be set to false.

Reply:
Basically when you run cluster test with `--run-once` and if test passes, we don't want to see those healthchecks

Reply:
Okay. Is there any scenario where we want to print health-checks even when the health-checks are passing?

If there isn't such a scenario, we can eliminate this parameter and only print health-checks when there are failures.

Reply:
We want health check to still print on `--healthcheck`...

Reply:
Thats actually probably only place where we want them all the time

Reply:
Updated.